### PR TITLE
GitHub Action to spellcheck and lint Python code

### PR DESCRIPTION
**Pull request by cclauss**
This Action uses minimal steps to run in ~5 seconds to rapidly:
* use `codespell` to look for typos in the codebase, and
* use `ruff` to lint Python code and provide intuitive GitHub Annotations to contributors.

Tools:
* https://github.com/codespell-project/codespell#readme
* https://docs.astral.sh/ruff

@RossBencina

### CODE REVIEW

File: test/patest_suggested_vs_streaminfo_latency.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def loadCsvData( dataFileName ):
     inputDevice = ""
     outputDevice = ""
 
-    startLines = file(dataFileName).readlines(1024)
+    startLines = open(dataFileName).readlines(1024)
```
Comment (cclauss) [inline test/patest_suggested_vs_streaminfo_latency.py:32]: The `from pylab import *` was hiding the `file()` issue that is just like:
* #898

**Comment by RossBencina** [comment]
I'd prefer to consider the python fixes and the CI workflow as separate PRs.

The fixes are uncontroversial and can me merged as is (although maybe an additional improvement is to explicitly scope the `pylab` symbols).

The CI I'm not sure about due to imposing linting opinions that I have no idea about since I've never used ruff or codespell.

**Comment by cclauss** [comment]
Removed the CI workflow.

**Review by philburk — APPROVED** [review]

**Review by RossBencina — APPROVED** [review]

**Comment by RossBencina** [comment]
Thank you!

**Pull request merged**